third_party/2and3/atomicwrites/__init__.pyi

```diff
@@ -1,13 +1,13 @@
-from typing import AnyStr, Callable, ContextManager, IO, Optional, Text, Type
+from typing import Any, AnyStr, Callable, ContextManager, Generic, IO, Optional, Text, Type, Union
 
 def replace_atomic(src: AnyStr, dst: AnyStr) -> None: ...
 def move_atomic(src: AnyStr, dst: AnyStr) -> None: ...
-class AtomicWriter(object):
-    def __init__(self, path: AnyStr, mode: Text = ..., overwrite: bool = ...) -> None: ...
-    def open(self) -> ContextManager[IO]: ...
-    def _open(self, get_fileobject: Callable) -> ContextManager[IO]: ...
-    def get_fileobject(self, dir: Optional[AnyStr] = ..., **kwargs) -> IO: ...
-    def sync(self, f: IO) -> None: ...
-    def commit(self, f: IO) -> None: ...
-    def rollback(self, f: IO) -> None: ...
-def atomic_write(path: AnyStr, writer_cls: Type[AtomicWriter] = ..., **cls_kwargs: object) -> ContextManager[IO]: ...
+class AtomicWriter(Generic[AnyStr]):
```

Comment:
Making this generic over AnyStr seems problematic because there's no AnyStr in the constructor signature, so mypy won't be able to infer the type variable if you create an AtomicWriter. I'm guessing the `mode` argument determines whether it becomes a bytes or a str writer, so maybe the right solution is to have overloads on `__new__` that return either an AtomicWriter[bytes] or an AtomicWriter[str] depending on the mode argument.

Reply:
That's fine with me.